perf: rename eventTypes.heavy to eventTypesHeavy

sean-brydon opened the pull request:
remove the mergeRotuer which after running a tsc trace - appears to be generating some nasty types. In favour of using a seperate router for this

sean-brydon committed:
Rename eventTypes.heavy to eventTypesHeavy

sean-brydon committed:
Update eventTypes lib types

keithwillcode commented on apps/web/app/_trpc/trpc-client.ts:
Delete this if. No longer needed

keithwillcode commented on packages/trpc/react/trpc.ts:
Delete this. No longer needed

sean-brydon committed:
Remove trpc and trpc client heavy resolve endpoint

sean-brydon committed:
Update shared file

sean-brydon committed:
Fix comments

keithwillcode reviewed (changes requested):
@sean-brydon Since this is going to now hit /api/trpc/eventTypesHeavy, you need to move that [trpc].ts file in app/web/pages/api/trpc/eventTypes/heavy/[trpc].ts

sean-brydon committed:
Rename path

sean-brydon committed:
Fix wait for URL path

keithwillcode committed:
Merge branch 'main' into perf/remove-merge-router-event-type-heavy

anikdhabal reviewed (changes requested):
E2E is failing, and it looks like we are missing that.

anikdhabal committed:
Fix API endpoint in event-types.e2e.ts test

anikdhabal committed:
Merge branch 'main' into perf/remove-merge-router-event-type-heavy

devin-ai-integration[bot] committed:
fix: Import WorkflowType from enums instead of client

The WorkflowType enum should be imported from @calcom/prisma/enums
rather than @calcom/prisma/client. This was causing integration tests
to fail with 'Cannot read properties of undefined (reading EVENT_TYPE)'
because the enum wasn't being properly exported from the client.

This fixes the remaining integration test failures after the Prisma 6.16
upgrade.

Co-Authored-By: anik@cal.com <adhabal2002@gmail.com>

anikdhabal reviewed (approved):
Looks Good
